testsuite/features/step_definitions/command_steps.rb

```diff
@@ -1576,3 +1577,117 @@
     node.run('mkinitrd')
   end
 end
+
+## ReportDB ##
+
+Given(/^I can connect to the ReportDB on the Server$/) do
+  # connect and quit database
+  _result, return_code = $server.run(reportdb_server_query('\\q'))
+  raise 'Couldn\'t connect to the ReportDB on the server' unless return_code.zero?
+end
+
+Given(/^I have a user with admin access to the ReportDB$/) do
+  users_and_permissions, return_code = $server.run(reportdb_server_query('\\du'))
+  raise 'Couldn\'t connect to the ReportDB on the server' unless return_code.zero?
+  # extract only the line for the suma user
+  suma_user_permissions = users_and_permissions[/pythia_susemanager(.*)}/]
+  raise 'ReportDB admin user pythia_susemanager doesn\'t have the required permissions' unless ['Superuser', 'Create role', 'Create DB'].all? { |permission| suma_user_permissions.include? permission }
+end
+
+When(/^I create a read-only user for the ReportDB$/) do
+  $reportdb_ro_user = 'test_user'
+  file = 'create_user_reportdb.exp'
+  source = "#{File.dirname(__FILE__)}/../upload_files/#{file}"
+  dest = "/tmp/#{file}"
+  return_code = file_inject($server, source, dest)
+  raise 'File injection in server failed' unless return_code.zero?
+  $server.run("expect -f /tmp/#{file} #{$reportdb_ro_user}")
+end
+
+Then(/^I should see the read-only user listed on the ReportDB user accounts$/) do
+  users_and_permissions, _code = $server.run(reportdb_server_query('\\du'))
+  raise 'Couldn\'t find the newly created user on the ReportDB' unless users_and_permissions.include? $reportdb_ro_user
+end
+
+When(/^I delete the read-only user for the ReportDB$/) do
+  file = 'delete_user_reportdb.exp'
+  source = "#{File.dirname(__FILE__)}/../upload_files/#{file}"
+  dest = "/tmp/#{file}"
+  return_code = file_inject($server, source, dest)
+  raise 'File injection in server failed' unless return_code.zero?
+  $server.run("expect -f /tmp/#{file} #{$reportdb_ro_user}")
+end
+
+Then(/^I shouldn't see the read-only user listed on the ReportDB user accounts$/) do
+  users_and_permissions, _code = $server.run(reportdb_server_query('\\du'))
+  raise 'Created read-only user on the ReportDB remains listed' if users_and_permissions.include? $reportdb_ro_user
+end
+
+When(/^I connect to the ReportDB with read-only user from external machine$/) do
+  # connection from the controller to the reportdb in the server
+  $reportdb_ro_conn = PG.connect(host: $server.public_ip, port: 5432, dbname: 'reportdb', user: $reportdb_ro_user, password: 'linux')
```

Comment:
I'm not sure if we want to let readers of Cucumber feature knows that the database name is "reportdb", maybe we need to expose it in the step, just an idea, not sure.

Reply:
not sure what you mean, sorry - do you mean accepting the db name as an argument in the step or saving it in a global variable to access here or...?